frontend/app/routes/public/apply/$id/terms-and-conditions.tsx

```diff
@@ -163,6 +249,28 @@ export default function ApplyIndex({ loaderData, params }: Route.ComponentProps)
           </ButtonLink>
         </div>
       </fetcher.Form>
+      <Dialog open={showDialog} onOpenChange={setShowDialog}>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>{t('apply:terms-and-conditions.dialog.title')}</DialogTitle>
+          </DialogHeader>
+          <DialogDescription>{t('apply:terms-and-conditions.dialog.description')}</DialogDescription>
+          <DialogFooter>
+            <DialogClose asChild>
+              <Button id="confirm-modal-back" disabled={isSubmitting} startIcon={faChevronLeft} variant="default" size="sm">
```

Comment:
Since it will be difficult to spot any missing Adobe Analytics attributes:

```suggestion
              <Button id="confirm-modal-back" disabled={isSubmitting} startIcon={faChevronLeft} variant="default" size="sm" data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form:Modal Back - Terms and Conditions click">
```